__tests__/warehouse.test.js

```diff
@@ -0,0 +1,196 @@
+const transformers = integrations.map(integration =>
+  require(`../${version}/destinations/${integration}/transform`)
+);
+const eventTypes = ["track", "identify", "page", "screen", "group", "alias"];
+
+describe("event types", () => {
+  describe("track", () => {
+    it("should generate two events for every track call", () => {
+      transformers.forEach((transformer, index) => {
+        const received = transformer.process(input("track"));
+        expect(received).toMatchObject(output("track", integrations[index]));
+      });
+    });
+  });
+
+  describe("identify", () => {
+    it("should generate two events for every identify call", () => {
+      // also verifies priority order between traits and context.traits
+      transformers.forEach((transformer, index) => {
+        const received = transformer.process(input("identify"));
+        expect(received).toMatchObject(output("identify", integrations[index]));
+      });
+    });
+  });
+
+  describe("page", () => {
+    it("should generate one events for every page call", () => {
+      transformers.forEach((transformer, index) => {
+        const received = transformer.process(input("page"));
+        expect(received).toMatchObject(output("page", integrations[index]));
+      });
+    });
+  });
+
+  describe("screen", () => {
+    it("should generate one events for every screen call", () => {
+      transformers.forEach((transformer, index) => {
+        const received = transformer.process(input("screen"));
+        expect(received).toMatchObject(output("screen", integrations[index]));
+      });
+    });
+  });
+
+  describe("alias", () => {
+    it("should generate one events for every alias call", () => {
+      transformers.forEach((transformer, index) => {
+        const received = transformer.process(input("alias"));
+        expect(received).toMatchObject(output("alias", integrations[index]));
+      });
+    });
+  });
+});
+
+describe("column & table names", () => {
+  it("should handle special character, spacing, snake case etc.", () => {
+    let i = input("track");
+
+    i.message.properties = Object.assign(
+      i.message.properties,
+      names.input.properties
+    );
+
+    transformers.forEach((transformer, index) => {
+      const received = transformer.process(i);
+
+      const provider =
+        integrations[index] === "snowflake" ? "snowflake" : "default";
+
+      expect(received[1].metadata.columns).toMatchObject(
+        names.output.columns[provider]
+      );
+      expect(received[1].data).toMatchObject(names.output.data[provider]);
+
+      for (let tableName in names.input.properties) {
+        i.message.event = tableName;
+        let out = transformer.process(i);
+        expect(out[1].metadata.table).toEqual(
+          names.output.namesMap[provider][tableName]
+        );
+      }
+    });
+  });
+});
+
+describe("conflict between rudder set props and user set props", () => {
```

Comment:
Definition of rudder set-prop vs user set-prop will help.

Reply:
added